### DISCUSSION TITLE
V14: Entity actions and modals

### DISCUSSION THREAD

**Discussion by enkelmedia** [umbraco/Umbraco-CMS#16527] [Features and ideas]
The modals and the flow for entity actions have been changed in Bellissima, moving the entity action modal from being close to the tree to showing up to the right.

Clicking an action in the tree:

![v14-org-tree-actions](https://github.com/umbraco/Umbraco-CMS/assets/1782524/f045d085-d7b0-4764-aeb9-b3aaa25e5576)

Will open the dialog as a modal to the right, I think this is unfortunate as it forces the editor to move both eyes and mouse over the whole screen to take the next step.

![v14-org-modal](https://github.com/umbraco/Umbraco-CMS/assets/1782524/f58aebe1-e0dd-47be-b149-bdfa18990a66)

I understand that this might be because it makes more sense when entity actions are used from a workspace view:

![umb-org-action-in-workspace-view](https://github.com/umbraco/Umbraco-CMS/assets/1782524/4a3ddb74-3889-4fe6-9d3e-2cb7e4f5e8be)

I see a couple of nice benefits with this new approach, one being that the modals can have different widths (great for e.g. "Culture and Hostnames".

However, I would like to bring some attention to this as I think that we could do this better.

Many entity actions are related to the entity's placement in the tree (move, sort, create, and especially "reload"). I would argue that it would make more sense to make the area where the entity actions are shown next to the tree some kind of new modal extension point so that it could look like this:

![v14-create-dialog](https://github.com/umbraco/Umbraco-CMS/assets/1782524/24e19359-be7c-47d8-8ff4-aee65f7bbdc8)

![v14-cutlure-and-host](https://github.com/umbraco/Umbraco-CMS/assets/1782524/de7b8096-7e50-4b74-9210-1278381373cd)

The idea is that there would be some kind of sizes for these modals as well. 

I would also argue that we need some way to indicate that a certain entity action only makes sense in the tree, to avoid showing the "reload" action in the workspace view.

It would be interesting to hear what other people think, maybe it's just me :)

**Comment by bjarnef** [comment]
I agree with this. Honestly I don't use the Actions menu in top right corner that much, but once in a while I do.
A newbie could probably think the Reload in Actions menu would reload workspace view.

Furthermore Culture & Hostnames mostly isn't used on daily basis and related to tree node, so I guess it could make sense if actions are aware of tree or workspace context?

An option could be that the overlay could open based on context, e.g. from right in actions menu and from left clicking tree actions.
